meshio/_helpers.py

```diff
@@ -242,23 +250,18 @@ def write(filename, mesh, file_format=None, **kwargs):
     :params point_data: Named additional point data to write to the file.
     :type point_data: dict
     """
-    path = pathlib.Path(filename)
-    if not file_format:
-        # deduce file format from extension
-        file_format = _filetype_from_path(path)
-
-    # check cells for sanity
-    for key, value in mesh.cells.items():
-        if key[:7] == "polygon":
-            if value.shape[1] != int(key[7:]):
-                raise WriteError()
-        elif key in num_nodes_per_cell:
-            if value.shape[1] != num_nodes_per_cell[key]:
-                raise WriteError()
-        else:
-            # we allow custom keys <https://github.com/nschloe/meshio/issues/501> and
-            # cannot check those
-            pass
+    if is_buffer(filename, "r"):
+        assert (
```

Comment:
assert? :smile_cat:

Reply:
Haha, started the changes before The Exceptioning, clearly didn't get all of them in the merge.